File: C/reverse_string.c
#include <stdio.h>
#include <string.h>

void reverseString(char *str, int n) {
    int left = 0, right = n - 1;
    while (left < right) {
        char temp = str[left];
        str[left] = str[right];
        str[right] = temp;
        left++;
        right--;
    }
}

int main() {
    char str[100];
    printf("Enter a string: ");
    scanf("%s", str);
    int n = strlen(str);
    reverseString(str, n);
    printf("Reversed string: %s\n", str);
    return 0;
}

File: Python/reverse_string.py
def reverse_string(s, n):
    s = list(s)
    left, right = 0, n-1
    while left < right:
        s[left], s[right] = s[right], s[left]
        left += 1
        right -= 1
    return ''.join(s)

if __name__ == "__main__":
    s = input("Enter a string: ")
    n = len(s)
    result = reverse_string(s, n)
    print(f"Reversed string: {result}")